File: BookingApp/src/main/java/com/example/BookingApp/renting/model/RentingItem.java
package com.example.BookingApp.renting.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;

import static javax.persistence.InheritanceType.TABLE_PER_CLASS;

@Entity
@JsonIgnoreProperties({"hibernateLazyInitializer"})
@Inheritance(strategy=TABLE_PER_CLASS)
@Getter
@Setter
@DiscriminatorColumn(name = "type", discriminatorType=DiscriminatorType.STRING)
public abstract class RentingItem {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO, generator="seq")
    @GenericGenerator(name = "seq", strategy="increment")
    private Long id;
    private String name;
    private String address;
    private String description;
    @Column(insertable = false, updatable = false)
    private String type;
    private int capacity;
}

File: BookingApp/frontend/src/app/shared/models/reservations/RentingItem.ts
import { AdditionalService } from "../AdditionalService";

export class RentingItem{
    public id : Number;
    public name : String;
    public address: String;
    public description : String;
    public capacity : Number;
    public additionalServices : AdditionalService[];
    public averageGrade : Number;
    public type : String;

    constructor(id : Number,name : String, address : String,description : String,capacity: Number){
        this.id = id;
        this.address = address;
        this.name = name;
        this.description = description;
        this.capacity = capacity;
    }

}